src/ui/selectBox.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+
+/*
+
+    Infusion > 4.0 compatible select box. Designed to work with a map of items.
+
+*/
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    // A select box component that relays changes in the selected value, but that has to be redrawn when the list of
+    // available options changes.
+    fluid.defaults("youme.selectBox", {
+        gradeNames: ["fluid.containerRenderingView"],
+
+        injectionType: "html",
+
+        selectors: {
+            label: ".youme-select-label",
+            selectBox: ".youme-select"
+        },
+
+        markup: {
+            // TODO: Discuss why the dom is only built correctly if there is an outer div wrapping all selectable content.
+            container: "<div class='youme-select-container'><h3 class='youme-select-label'>%label</h3><select class='youme-select'>%optionsContent</select></div>",
+            optionItem: "<option value='%id' %selected>%name</option>\n"
+        },
+
+        includeNone: true,
+        noneLabel: "Select an Option",
+
+        model: {
+            label: "",
+            selectedItemId: false,
+            optionItems: {}
+        },
+
+        invokers: {
+            handleSelectChange: {
+                funcName: "youme.selectBox.handleSelectChange",
+                args: ["{that}", "{arguments}.0"]
+            },
+            renderMarkup: {
+                funcName: "youme.selectBox.render",
+                args: ["{that}", "{that}.options.markup.container"] // containerTemplate
+            },
+            renderOptions: {
+                funcName: "youme.selectBox.renderOptions",
+                args: ["{that}", "{that}.options.markup.optionItem" ] // itemTemplate
+            }
+        },
+        listeners: {
+            "onCreate.bind": {
+                this: "{that}.dom.selectBox",
+                method: "change",
+                args: ["{that}.handleSelectChange"]
+            }
+        },
+        modelListeners: {
+            selectBoxRefresh: {
+                excludeSource: "init",
+                path: ["selectedItemId", "optionItems"],
+                this: "{that}.dom.selectBox",
+                method: "html",
+                args: ["@expand:{that}.renderOptions()"]
+            },
+            label: {
+                excludeSource: "init",
+                this: "{that}.dom.label",
+                method: "html",
+                args: ["{that}.model.label"]
+            }
+        }
+    });
+
+    youme.selectBox.renderOptions = function (that, itemTemplate) {
+        var optionsContent = "";
+
+        if (that.options.includeNone) {
+            optionsContent += fluid.stringTemplate(itemTemplate, { id: "none", name: that.options.noneLabel });
```

Comment:
Again another note to self - given I find myself writing a lot of this kind of stuff these days too, it's seeming more and more plausible that Infusion 6 will render based on some embedded copy of preact.
Note that preact also now bundles a more lightweight "HTM" format for templates that we could support directly in place of the component's template option: https://preactjs.com/guide/v10/getting-started/#alternatives-to-jsx

Reply:
If you like it, should be easy enough to create an override of renderContainer https://github.com/fluid-project/infusion/blob/main/src/framework/core/js/NewViewSupport.js#L152 which uses HTM bound to an h( .... ) function which creates real DOM nodes rather than fake ones - https://github.com/developit/htm#advanced-usage

Reply:
I have been thinking a lot about this, but would like to do this as another small package to avoid having my non-music stuff (gamepad navigator, etc.) have to depend on this package.